{"uuid":"0465d8e5-bcc7-4384-b6a0-33b4f6589b2c","historyId":"b40194f764b47c372945af488930df1c","testCaseId":"48e1e669a8a0dd2cef8938296d4ea7a9","fullName":"src/test/resources/features/DeleteBook.feature:4","labels":[{"name":"tag","value":"DELETEBook"},{"name":"host","value":"laptop"},{"name":"thread","value":"2268@laptop.main(1)"},{"name":"feature","value":"Delete a book from the system"},{"name":"story","value":"User attempts to delete a book with a valid ID"},{"name":"suite","value":"Delete a book from the system"},{"name":"testClass","value":"User attempts to delete a book with a valid ID"},{"name":"framework","value":"cucumber4jvm"},{"name":"language","value":"java"},{"name":"gherkin_uri","value":"file:///E:/IT%20%20NOTES/IS3440%20-%20IT%20Quality%20Assurance/Group%2029/API_Testing/src/test/resources/features/DeleteBook.feature"},{"name":"package","value":"src.test.resources.features.DeleteBook_feature.Delete a book from the system"}],"links":[],"name":"User attempts to delete a book with a valid ID","status":"failed","statusDetails":{"known":false,"muted":false,"flaky":false,"message":"Unexpected status code! This is a BUG. expected [403] but found [404]","trace":"java.lang.AssertionError: Unexpected status code! This is a BUG. expected [403] but found [404]\r\n\tat org.testng.Assert.fail(Assert.java:111)\r\n\tat org.testng.Assert.failNotEquals(Assert.java:1578)\r\n\tat org.testng.Assert.assertEqualsImpl(Assert.java:150)\r\n\tat org.testng.Assert.assertEquals(Assert.java:132)\r\n\tat org.testng.Assert.assertEquals(Assert.java:1419)\r\n\tat org.testng.Assert.assertEquals(Assert.java:1383)\r\n\tat com.group_29.api_testing.stepDefinitions.deleteApiStepDefinitions.DeleteBooksTest.iShouldReceiveAResponseWithStatusCode(DeleteBooksTest.java:74)\r\n\tat ✽.I should receive a delete response with status code 403(file:///E:/IT%20%20NOTES/IS3440%20-%20IT%20Quality%20Assurance/Group%2029/API_Testing/src/test/resources/features/DeleteBook.feature:7)\r\n"},"stage":"finished","steps":[{"name":"Given I am a user in delete","status":"passed","statusDetails":{"known":false,"muted":false,"flaky":false},"stage":"finished","steps":[],"attachments":[],"parameters":[],"start":1735708014264,"stop":1735708014600},{"name":"When I send a DELETE request to \"/api/books/{id}\" with a valid book ID as a user","status":"passed","statusDetails":{"known":false,"muted":false,"flaky":false},"stage":"finished","steps":[],"attachments":[],"parameters":[],"start":1735708014601,"stop":1735708015688},{"name":"Then I should receive a delete response with status code 403","status":"failed","statusDetails":{"known":false,"muted":false,"flaky":false,"message":"Unexpected status code! This is a BUG. expected [403] but found [404]","trace":"java.lang.AssertionError: Unexpected status code! This is a BUG. expected [403] but found [404]\r\n\tat org.testng.Assert.fail(Assert.java:111)\r\n\tat org.testng.Assert.failNotEquals(Assert.java:1578)\r\n\tat org.testng.Assert.assertEqualsImpl(Assert.java:150)\r\n\tat org.testng.Assert.assertEquals(Assert.java:132)\r\n\tat org.testng.Assert.assertEquals(Assert.java:1419)\r\n\tat org.testng.Assert.assertEquals(Assert.java:1383)\r\n\tat com.group_29.api_testing.stepDefinitions.deleteApiStepDefinitions.DeleteBooksTest.iShouldReceiveAResponseWithStatusCode(DeleteBooksTest.java:74)\r\n\tat ✽.I should receive a delete response with status code 403(file:///E:/IT%20%20NOTES/IS3440%20-%20IT%20Quality%20Assurance/Group%2029/API_Testing/src/test/resources/features/DeleteBook.feature:7)\r\n"},"stage":"finished","steps":[],"attachments":[],"parameters":[],"start":1735708015688,"stop":1735708015702}],"attachments":[],"parameters":[],"start":1735708014262,"stop":1735708015703}